Source/Core/DolphinQt/GameList/GameListModel.cpp

```diff
@@ -140,7 +142,26 @@ QVariant GameListModel::data(const QModelIndex& index, int role) const
     break;
   case COL_FILE_NAME:
     if (role == Qt::DisplayRole || role == Qt::InitialSortOrderRole)
+    {
+      // If the game is extracted (ex. the file is named main.dol), use the
+      // name of the folder that is two directories up
+      if (game.GetFileName() == "main.dol")
+      {
+        std::filesystem::path game_file_path = game.GetFilePath();
```

Comment:
I believe this will cause encoding errors when using UTF-8 paths on Windows. You should use this instead: https://github.com/dolphin-emu/dolphin/blob/0461170363693efec7d6a8a34baaa5fb2deb8954/Source/Core/Common/StringUtil.h#L206